labs/waterfall.md

```diff
@@ -0,0 +1,55 @@
+---
+layout: default
+title: Waterfall Network Proposal
+parent: Labs
+---
+# Lab Name
+[Waterfall Network](https://waterfall.network/)
+
+# Short Description
+Waterfall is a highly scalable smart contract network for the development of various decentralized applications (DApps).
```

Comment:
Thanks for the proposal. Need more clarity on the protocol, few questions I have:
1) What sort of protocol is it ? Please elaborate how does it integrate with EVM.
2) Does the smart contract transaction execution happen on its own network or the Layer 1 network ?
3) Does it support Non EVM use cases as well ?
4) The Dapps that the user develops using Waterfall can interact with which blockchain network ? Or is it like Waterfall network is the Layer one itself ?

Reply:
Thank you for the good questions: 

1. It is a layer one network with its own state and security. It uses actual EVM(not any sort of emulation), and on the level of EVM everything works identically to Ethereum - EVM receives an ordered set of transactions for execution, executes them, and the network state is updated. 

2. Everything happens on Waterfall own layer 1 network.  

3. Not at this stage, but the modularity of the virtual machine and its expansion beyond the scope of EVM were discussed and estimated, but in the team's opinion, they are not priorities as of now. 

4. Waterfall is layer one itself, but it is already connected through bridges to other EVM-based networks, s.a. Base, Polygon, Arbitrum, BSC, and later to Bitcoin and other non-EVM networks.

Reply:
Thanks a lot for the responses. Great proposal! I will approve it!

Reply:
Thank you!